detox/android/detox/src/full/java/com/wix/detox/reactnative/idlingresources/UIModuleIdlingResource.java

```diff
@@ -122,6 +140,33 @@ protected boolean checkIdle() {
         return true;
     }
 
+    // This is a workaround for https://github.com/facebook/react-native/issues/32594
+    // uses duck typing heuristics to determine that this is probably the stuck Switch operation and if so, ignores it
+    private boolean workaroundForRN66Bug(Object uiOperationQueue) {
```

Comment:
This must be extracted onto a separate class/module. We cannot keep being affected by the broken-window effect of these legacy-code grade classes. Here's one example as to how this can be achieved: https://github.com/wix/Detox/blob/d95d924ae323edfab832f281b7713f6519139d0d/detox/android/detox/src/full/java/com/wix/detox/reactnative/idlingresources/SerialExecutorReflected.kt#L6

Reply:
Double ping :-)